src/node.cpp

```diff
@@ -19,35 +20,102 @@ namespace alice {
               << " with message type " << static_cast<int>(packet.type) << std::endl;
     }
 
-    void Node::receivePacket(const std::vector<uint8_t>& data) const {
-        Packet packet = Packet::deserialize(data);
-        processPacket(packet);
+    void Node::sendPacket(const Packet& packet) {
+        Packet packet_to_send = packet;
+        packet_to_send.sequence_number = next_sequence_number_++;
+        packet_to_send.timestamp = static_cast<uint64_t>(std::time(nullptr));
+
+        std::vector<uint8_t> data = packet_to_send.serialize(encryptor_);
```

Comment:
where do you get  encryptor_ variable from? I don't think I see it in ur interface or declared anywhere else. in this commit.